### PR TITLE
Remove test warnings again, minimize output, hide more async stuff

### PR DESCRIPTION
**Pull request by tlambert03**
# Description
with napari-svg 0.1.5, ~this PR makes *all* warnings errors now~, (unless explicitly ignored, as usual, in pyproject.toml).  It also turns off verbose mode and adds `--maxfail=5`  to make our logs a bit easier to parse on github.

**edit**:  scratch that warnings are still only errors if internal to napari: `"error:::napari"`

@carreau, I deleted what seemed to be duplicate pytest options in `setup.cfg`  (since we use the one in `pyproject.toml`) is that ok?

As a reminder... Deprecation and FutureWarnings, should not be "ignored" in `filterwarnings` in pyproject.toml.  Instead, they should be explicitly asserted in the relevant tests (such that the test will fail when the deprecation warning has been removed, so we can clean it up at that time).

## Type of change

- [x] tests

# References

# How has this been tested?

- [ ] example: the test suite for my feature covers cases x, y, and z
- [ ] example: all tests pass with my change

## Final checklist:
- [ ] My PR is the minimum possible work for the desired functionality
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] If I included new strings, I have used `trans.` to make them localizable.
      For more information see our [translations guide](https://napari.org/docs/dev/guides/translations.html).

### CODE REVIEW

**Review by goanpeca — APPROVED** [review]

**Comment by tlambert03** [comment]
we're also starting to get a lot of random segfaults in even our non-async our tests related to the `chunk/_delay_queue.py` (See [here](https://github.com/napari/napari/pull/2648/checks?check_run_id=2488921726#step:7:2561)). 

I'm going to make a few more things `async_only` in this PR, but  I'd like us to go through all of our tests again sometime soon and make sure that all of that async stuff is actually opt-in.  It continues to be a big source of test problems

File: napari/_qt/qt_viewer.py [hunk 1]
```diff
@@ -288,7 +289,8 @@ def _create_canvas(self) -> None:
         theme.connect(on_theme_change)
 
         def disconnect():
-            theme.disconnect(on_theme_change)
+            with suppress(AttributeError):
```
Comment (sofroniewn) [inline napari/_qt/qt_viewer.py:294]: Can you maybe add a comment here on why there might be an `AttributeError`

File: pyproject.toml [hunk 2]
```diff
@@ -73,18 +73,23 @@ skip_glob = ["*examples/*", "*vendored*", "*_vendor*"]
 [tool.pytest.ini_options]
 # These follow standard library warnings filters syntax.  See more here:
 # https://docs.python.org/3/library/warnings.html#describing-warning-filters
-addopts = "--durations=10"
+addopts = "--maxfail=5 --durations=5"
+
+# NOTE: only put things that will never change in here.
```
Comment (sofroniewn) [inline pyproject.toml:78]: Nice!

**Review by sofroniewn — APPROVED** [review]
LGTM! Left a minor comment or two, but let's merge once tests pass!!

**Comment by sofroniewn** [comment]
> I'm going to make a few more things async_only in this PR, but I'd like us to go through all of our tests again sometime soon and make sure that all of that async stuff is actually opt-in. It continues to be a big source of test problems

Yes, we can continue to do this after this PR too

File: napari/_qt/qt_viewer.py (hunk 1 above)
Comment (tlambert03) [reply napari/_qt/qt_viewer.py:294]: I honestly have no idea... i think it has something to do with cleanup order?  Will add a note about it being mysterious

Comment (sofroniewn) [reply napari/_qt/qt_viewer.py:294]: lol, ok, nice - a note about it being mysterious is better than no note at all :-)

**Pull request merged**